src/replay_syscall.cc

```diff
@@ -673,8 +673,17 @@ static void finish_shared_mmap(ReplayTask* t, AutoRemoteSyscalls& remote,
   // For full generality, we also need to emulate direct file
   // modifications through write/splice/etc.
   off64_t offset_bytes = page_size() * offset_pages;
-  if (ssize_t(buf.data.size()) !=
-      pwrite64(emufile->fd(), buf.data.data(), buf.data.size(), offset_bytes)) {
+  ssize_t res = 0;
+  do
+  {
+    ssize_t c = pwrite64(emufile->fd(), buf.data.data() + res, buf.data.size() - res, offset_bytes + res);
+    if (c == -1)
+    break;
```

Comment:
If `pwrite64` returns 0 then I guess we need to abort because the device is full...